springboot/src/main/java/ca/jrvs/apps/trading/dao/MarketDataDao.java

```diff
@@ -0,0 +1,183 @@
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataRetrievalFailureException;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class MarketDataDao implements CrudRepository<IexQuote, String> {
+
+  private static final String IEX_BATCH_PATH = "stock/market/batch?symbols=%s&types=quote&token=";
+  private final String IEX_BATCH_URL;
+
+  private Logger logger = LoggerFactory.getLogger(MarketDataDao.class);
+  private HttpClientConnectionManager httpClientConnectionManager;
+
+  @Autowired
+  public MarketDataDao(HttpClientConnectionManager httpClientConnectionManager,
+      MarketDataConfig marketDataConfig) {
+    this.httpClientConnectionManager = httpClientConnectionManager;
+    IEX_BATCH_URL = marketDataConfig.getHost() +  IEX_BATCH_PATH + marketDataConfig.getToken();
+  }
+
+
+  @Override
+  public <S extends IexQuote> S save(S ticker) {
+    throw new UnsupportedOperationException("Not implemented");
+  }
+
+  @Override
+  public <S extends IexQuote> Iterable<S> saveAll(Iterable<S> tickers) {
+    throw new UnsupportedOperationException("Not implemented");
+  }
+
+  /**
+   * Get an IexQuote (helper method which class findAllById)
+   *
+   * @param ticker
+   * @throws IllegalArgumentException if a given ticker is invalid
+   * @throws DataRetrievalFailureException if HTTP request failed
+   */
+  @Override
+  public Optional<IexQuote> findById(String ticker) {
+    Optional<IexQuote> iexQuote;
+    List<IexQuote> quotes = findAllById(Collections.singletonList(ticker));
+
+    if (quotes.size() == 0) {
+      return Optional.empty();
+    } else if (quotes.size() == 1) {
+      iexQuote = Optional.of(quotes.get(0));
+    } else {
+      throw new DataRetrievalFailureException("Unexpected number of quotes");
+    }
+    return iexQuote;
+  }
+
+  @Override
+  public boolean existsById(String ticker) {
+    throw new UnsupportedOperationException("Not implemented");
+  }
+
+  @Override
+  public Iterable<IexQuote> findAll() {
+    throw new UnsupportedOperationException("Not implemented");
+  }
+
+  /**
+   * Get quotes from IEX
+   * @param tickers is a ist of tickers
+   * @throws IllegalArgumentException if ticker is invalid or tickers is emtpy
+   * @throws DataRetrievalFailureException if HTTP request failed
+   */
+  @Override
+  public List<IexQuote> findAllById(Iterable<String> tickers) {
+    if (((Collection<?>) tickers).isEmpty()) {
+      throw new IllegalArgumentException("ERROR: List of tickers is empty");
+    }
+    String url = String.format(IEX_BATCH_URL, String.join(",", tickers));
+    Optional<String> response = executeHttpGet(url);
+    List<IexQuote> quotes = new ArrayList<>();
+    JSONObject jsonObject = null;
+    try {
+      jsonObject = new JSONObject(response.get());
+    } catch (JSONException e) {
+      e.printStackTrace();
```

Comment:
never use printstacktrace. use logger instead. please review the java coding style